### ISSUE TITLE
One or more environment variables failed assertions: ... 

### ISSUE THREAD

**Issue by OnkelTem** [vlucas/phpdotenv#524]
A use a third-party project which fails to run, because it cannot find variables from my `.env` file.

I found I code like this:

```php
$dotenv = Dotenv::createImmutable( __DIR__ . '/../' );
$dotenv->required( [ 'WPC_DB_HOST', 'WPC_DB_NAME', 'WPC_DB_USER', 'WPC_DB_PASSWORD' ] );
$dotenv->ifPresent( 'ALLOW_INSECURE' )->isBoolean();
$dotenv->load();
```
which raises exception in the second line, so the `load()` method has no chance to be called.

I wonder why it was supposed to work?

**Comment by GrahamCampbell** [comment]
That's correct. You need to move your validation to **after** the call to load. The validation immediately performs assertions about the current state of your environment, not about the env file. :)

**Comment by OnkelTem** [comment]
Hehe, I can't imagine how it worked in the original code :)
